java/datastore/src/main/java/com/google/datastore/v1/client/RemoteRpc.java

```diff
@@ -110,6 +127,22 @@ public InputStream call(String methodName, MessageLite request) throws Datastore
     }
   }
 
+  @VisibleForTesting
+  void setHeaders(MessageLite request, HttpRequest httpRequest) {
+    httpRequest.getHeaders().put(API_FORMAT_VERSION_HEADER, API_FORMAT_VERSION);
+    if (enableE2EChecksum && request != null) {
+      byte[] checksum = EndToEndChecksumHandler.computeChecksum(request.toByteArray());
+      if (checksum != null) {
+        httpRequest.getHeaders().put(EndToEndChecksumHandler.HTTP_REQUEST_CHECKSUM_HEADER,
+            new String(checksum));
```

Comment:
We should probably specify the character set on this constructed string to avoid any auto-detection.

Reply:
Done